File: examples/python/sdk-reference/serialization/filesystem-serdes-walkthrough.py
from aws_durable_execution_sdk_python import (
    DurableContext,
    StepContext,
    durable_execution,
    durable_step,
)
from aws_durable_execution_sdk_python.config import StepConfig
from aws_durable_execution_sdk_python.filesystem_serdes import FileSystemSerDes


@durable_step
def fetch_order(ctx: StepContext) -> dict:
    return {"id": "order-123", "total": "99.99"}


@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    fs_serdes = FileSystemSerDes("/mnt/s3")

    # Pass the FileSystem serdes to one step. Other operations in this handler
    # keep using the default serdes.
    order = context.step(
        fetch_order(),
        config=StepConfig(serdes=fs_serdes),
    )

    return order

File: examples/java/sdk-reference/serialization/filesystem-serdes-walkthrough.java
import java.nio.file.Path;
import java.util.Map;
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;
import software.amazon.lambda.durable.StepContext;
import software.amazon.lambda.durable.config.StepConfig;
import software.amazon.lambda.durable.serde.JacksonSerDes;
import software.amazon.lambda.durable.serde.SerDes;
import software.amazon.lambda.durable.serde.filesystem.FileSystemSerDesStage;

public class FileSystemSerDesWalkthrough
        extends DurableHandler<Object, Map<String, String>> {
    @Override
    public Map<String, String> handleRequest(Object input, DurableContext context) {
        SerDes fileSystemSerDes = new JacksonSerDes()
            .then(FileSystemSerDesStage.builder(Path.of("/mnt/s3")).build());

        StepConfig config = StepConfig.builder()
            .serDes(fileSystemSerDes)
            .build();

        // Pass the pipeline to one step. Other operations in this handler keep
        // using the default Jackson SerDes.
        return context.step(
            "fetch-order",
            Map.class,
            (StepContext ctx) -> Map.of("id", "order-123", "total", "99.99"),
            config
        );
    }
}
